elliottlib/errata_async.py

```diff
@@ -0,0 +1,116 @@
+
+    async def get_cve_package_exclusions(self, advisory_id: int):
+        path = f"/api/v1/cve_package_exclusion?filter[errata_id]={int(advisory_id)}"
+        # This is a paginated API, we need to increment page[number] until an empty array is returned.
+        params = {"page[number]": 1}
+        while True:
+            result = await self._make_request(aiohttp.hdrs.METH_GET, path, params=params)
+            data: List[Dict] = result.get('data', [])
+            if not data:
+                break
+            for item in data:
+                yield item
+            params["page[number]"] += 1
+
+    async def create_cve_package_exclusion(self, advisory_id: int, cve: str, package: str):
+        path = "/api/v1/cve_package_exclusion"
+        data = {
+            "cve": cve,
+            "errata": advisory_id,
+            "package": package,
+        }
+        return await self._make_request(aiohttp.hdrs.METH_POST, path, json=data)
+
+    async def delete_cve_package_exclusion(self, exclusion_id: int):
+        path = f"/api/v1/cve_package_exclusion/{int(exclusion_id)}"
+        await self._make_request(aiohttp.hdrs.METH_DELETE, path, parse_json=False)
+
+
+class AsyncErrataUtils:
+    @classmethod
+    async def compare_cve_package_exclusions(cls):
+        return None, None
+
+    @classmethod
+    async def associate_builds_with_cves(cls, api: AsyncErrataAPI, advisory_id: int, attached_builds: List[Dict], cve_components_mapping: Dict[str, Set[str]]):
+        brew_components = {parse_nvr(nvr)["name"] for nvr in attached_builds}
+        expected_cve_packages_exclusions = {
+            cve: brew_components - components
+            for cve, components in cve_components_mapping.items()
+        }
+        _LOGGER.debug("Getting associated CVEs for advisory %s", advisory_id)
+        advisory_cves = await api.get_cves(advisory_id)
+        extra_cves = cve_components_mapping.keys() - advisory_cves
+        if extra_cves:
+            raise ValueError(f"The following CVEs are not associated with advisory {advisory_id}: {', '.join(sorted(extra_cves))}")
+
+        _LOGGER.debug("Creating current CVE exclusions for advisory %s", advisory_id)
+        current_cve_package_exclusions = {}
+        async for cve_package_exclusion in api.get_cve_package_exclusions(advisory_id):
+            current_cve_package_exclusions.setdefault(cve_package_exclusion["relationships"]["cve"]["name"], {})[cve_package_exclusion["relationships"]["package"]["name"]] = cve_package_exclusion["id"]
+
+        extra_cve_package_exclusions = {cve: exclusions.keys() - expected_cve_packages_exclusions.get(cve, set()) for cve, exclusions in current_cve_package_exclusions.items()}
+        missing_cve_package_exclusions = {cve: exclusions - current_cve_package_exclusions.get(cve, {}).keys() for cve, exclusions in expected_cve_packages_exclusions.items()}
+
+        _LOGGER.debug("Reconsiling CVE package exclusions for advisory %s", advisory_id)
```

Comment:
Could the loglevel for application logs be `info`?
Part of the reason why I think that'd be nice is to not accidentally get a debug level logging for http requests, which might leak kerberos tokens.

Reply:
Also, `s/Reconsiling/Reconciling/` :)